pkg/credentialprovider/external_config.go

```diff
@@ -0,0 +1,43 @@
+package credentialprovider
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/registrycredential"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/registrycredential/v1alpha1"
+)
+
+func readExternalProviderConfig(configPath string) (*registrycredential.RegistryCredentialConfig, error) {
+	if configPath == "" {
+		klog.Warningf("RegistryCredentialConfigPath is empty, no external registry credential providers will be configured.")
+		return nil, nil
```

Comment:
if I have the feature gate enabled but leave RegistryCredentialConfigPath empty, I'll be left with no credentials provider at all. Is that intended or should it fall back to using an internal provider?

Reply:
I was not intending for that scenario to work, as I thought its more trouble than its worth.  Do you think its necessary?  I'd rather like to see what exactly works and what doesn't when the gate is enabled.

Reply:
No, it's fine to leave it as is. But it definitely needs to be revisited when the feature moves to beta and becomes on by default, maybe leave a TODO.

Reply:
We should fix this now. People deploy clusters with AllAlpha=true. We run tests like this. We shouldn't break everyone who uses this configuration and doesn't have a credential provider.

Reply:
I've added logic to separate enabling the feature gate and enabling the feature:
```
if utilfeature.DefaultFeatureGate.Enabled(features.ExternalRegistryCredentialProviders) 
        && registryCredentialConfigPath != "" {
    kubeRuntimeManager.keyring, err = credentialprovider.NewExternalProviderKeyring(registryCredentialConfigPath)
    if err != nil {
        klog.Errorf("Failed to read external registry credential provider config.  Kubelet may not be able to pull images: %v", err)
    }
} else {
    if utilfeature.DefaultFeatureGate.Enabled(features.ExternalRegistryCredentialProviders) 
            && registryCredentialConfigPath == "" {
        klog.Warningln("The feature gate ExternalRegistryCredentialProviders is enabled, but RegistryCredentialConfigPath is not set.  Using legacy credential provider.")
    }
    kubeRuntimeManager.keyring = credentialprovider.NewDockerKeyring()
}

```